>From aa9f7162da91932fdc76aeeb6cd7ab6adcbdc9b9 Mon Sep 17 00:00:00 2001
From: Michael Walle <michael@walle.cc>
Date: Thu, 10 Apr 2014 13:55:36 +0200
Subject: [PATCH] Add support for Buffalo Linkstation LS-CHLv2/LS-XHL.

Signed-off-by: Michael Walle <michael@walle.cc>
---
 build/config/armel/kirkwood/network-console.cfg |   22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/build/config/armel/kirkwood/network-console.cfg b/build/config/armel/kirkwood/network-console.cfg
index 24f8803..307fb2c 100644
--- a/build/config/armel/kirkwood/network-console.cfg
+++ b/build/config/armel/kirkwood/network-console.cfg
@@ -1,9 +1,11 @@
 MEDIA_TYPE = netboot image (SSH)
 
-TARGET = $(TEMP_INITRD) $(TEMP_KERNEL) ts219 ts41x lacie
+TARGET = $(TEMP_INITRD) $(TEMP_KERNEL) ts219 ts41x lacie lschlv2 lsxhl
 EXTRANAME = $(MEDIUM)
 TYPE = netboot/network-console
 
+TEMP_DTB = $(TEMP)/lib
+
 # QNAP TS-119, TS-219 and TS-219P
 ts219:
 	mkdir -p $(SOME_DEST)/$(EXTRANAME)/qnap/ts-219
@@ -47,3 +49,21 @@ lacie:
 	mkimage -A arm -O linux -T ramdisk -C gzip -a 0x0 -e 0x0 -n "debian-installer ramdisk" -d $(TEMP_INITRD) $(SOME_DEST)/$(EXTRANAME)/lacie/uInitrd
 	update-manifest $(SOME_DEST)/$(EXTRANAME)/lacie/uImage "Linux kernel for LaCie NAS"
 	update-manifest $(SOME_DEST)/$(EXTRANAME)/lacie/uInitrd "initrd for LaCie NAS"
+
+# Buffalo Linkstation LS-CHLv2
+lschlv2:
+	mkdir -p $(SOME_DEST)/$(EXTRANAME)/buffalo/lschlv2
+	cat $(TEMP_KERNEL) $(TEMP_DTB)/kirkwood-lschlv2.dtb > $(TEMP)/vmlinuz-lschlv2
+	mkimage -A arm -O linux -T kernel -C none -a 0x00008000 -e 0x00008000 -n "Debian kernel" -d $(TEMP)/vmlinuz-lschlv2 $(SOME_DEST)/$(EXTRANAME)/buffalo/lschlv2/uImage
+	mkimage -A arm -O linux -T ramdisk -C gzip -a 0x0 -e 0x0 -n "debian-installer ramdisk" -d $(TEMP_INITRD) $(SOME_DEST)/$(EXTRANAME)/buffalo/lschlv2/uInitrd
+	update-manifest $(SOME_DEST)/$(EXTRANAME)/buffalo/lschlv2/uImage "Linux kernel for LS-CHLv2"
+	update-manifest $(SOME_DEST)/$(EXTRANAME)/buffalo/lschlv2/uInitrd "initrd for LS-CHLv2"
+
+# Buffalo Linkstation LS-XHL
+lsxhl: lschlv2
+	mkdir -p $(SOME_DEST)/$(EXTRANAME)/buffalo/lsxhl
+	cat $(TEMP_KERNEL) $(TEMP_DTB)/kirkwood-lsxhl.dtb > $(TEMP)/vmlinuz-lsxhl
+	mkimage -A arm -O linux -T kernel -C none -a 0x00008000 -e 0x00008000 -n "Debian kernel" -d $(TEMP)/vmlinuz-lsxhl $(SOME_DEST)/$(EXTRANAME)/buffalo/lsxhl/uImage
+	ln -nfs ../lschlv2/uInitrd $(SOME_DEST)/$(EXTRANAME)/buffalo/lsxhl/uInitrd
+	update-manifest $(SOME_DEST)/$(EXTRANAME)/buffalo/lsxhl/uImage "Linux kernel for LS-XHL"
+	update-manifest $(SOME_DEST)/$(EXTRANAME)/buffalo/lsxhl/uInitrd "initrd for LS-XHL"
-- 
1.7.10.4

